src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -371,16 +368,62 @@ private async ValueTask<EvaluationEvent> EvaluateFeature<TContext>(string featur
                     await sessionManager.SetAsync(evaluationEvent.FeatureDefinition.Name, evaluationEvent.Enabled).ConfigureAwait(false);
                 }
 
-                if (evaluationEvent.FeatureDefinition.Telemetry != null &&
-                    evaluationEvent.FeatureDefinition.Telemetry.Enabled)
+                // Only add an activity event if telemetry is enabled for the feature
+                if (telemetryEnabled)
                 {
-                    PublishTelemetry(evaluationEvent, cancellationToken);
+                    AddEvaluationActivityEvent(evaluationEvent);
                 }
             }
 
             return evaluationEvent;
         }
 
+        private void AddEvaluationActivityEvent(EvaluationEvent evaluationEvent)
+        {
+            // Break if there is no reason to emit an event
+            if (Activity.Current == null || !Activity.Current.IsAllDataRequested)
+            {
+                return;
+            }
+
+            ActivityTagsCollection tags = new ActivityTagsCollection
+            {
+                { "FeatureName", evaluationEvent.FeatureDefinition.Name },
+                { "Enabled", evaluationEvent.Enabled },
+                { "VariantAssignmentReason", evaluationEvent.VariantAssignmentReason },
+                { "Version", ActivitySource.Version }
+            };
+
+            if (evaluationEvent.TargetingContext is { UserId: string userId })
+            {
+                tags["TargetingId"] = userId;
+            }
+
+            if (evaluationEvent.Variant is { Name: string variantName })
+            {
+                tags["Variant"] = variantName;
+            }
+
+            if (evaluationEvent.FeatureDefinition.Telemetry.Metadata is { } metadata)
```

Comment:
Why do we need anything more than a null-check here? We already know the type of `Metadata`.

Reply:
> Why do we need anything more than a null-check here? 

We don't. That line _is_ an inverted null check, but it extracts the value into the `metatada` variable so you don't need to repeat the entire thing again inside the block.

> We already know the type of `Metadata`.

Yes, which is why we use `is { } metadata` instead of `is TheTypeOfTheMetadata metadata`. Using `{ }` is a way to avoid needing to specify the type.

It would basically be equivalent to:
```csharp
if (evaluationEvent.FeatureDefinition.Telemetry.Metadata is not null and var metadata)
```

Reply:
`is` is good for type checking. I'd avoid introducing usage of `is` here now to replace basic null checks.